modules/45_carbonprice/diffExp2Lin/datainput.gms

```diff
@@ -47,15 +46,16 @@ display p45_phasein_2025ratio;
 *** according to Asian Modeling Excercise tax case setup, 30$/t CO2eq in 2020 = 0.110 k$/tC
 
 *** for the current implementation, use the following trajectory for rich countries:
-*** price increases exponentially from cm_co2_tax_2020 in 2020
-if(cm_co2_tax_2020 lt 0,
-abort "please choose a valid cm_co2_tax_2020"
-elseif cm_co2_tax_2020 ge 0,
+*** price increases exponentially with growth rate cm_co2_tax_growth from cm_co2_tax_startyear in startyear
+*** for peak budget runs (if cm_iterative_target_adj = 6|7|9), the adjustment of the CO2 price trajectory after the peak year is made in core/postsolve.gms
+if(cm_co2_tax_startyear le 0,
+  abort "please choose a valid cm_co2_tax_startyear"
+elseif cm_co2_tax_startyear gt 0,
 *** convert tax value from $/t CO2eq to T$/GtC
-p45_CO2priceTrajDeveloped("2020") = cm_co2_tax_2020 * sm_DptCO2_2_TDpGtC;
```

Comment:
do we still need cm_co2_tax_2020?

Reply:
Yes, it is used in the realizations `diffCurvPhaseIn2Lin`, `expoLinear`, `exponential`, and `linear`. I plan to delete all these realizations except  `expoLinear` because there functionalities are covered by `diffExp2Lin `and `diffLin2Lin`. Then I can also delete this switch. But I would like to wait with the clean up until my next PR (which will have further developments on diffLin2Lin and will come in August after my vacation).

Reply:
I agree and thanks for cleaning up